src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,35 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executor = Executors.newCachedThreadPool();
 
+    public void registerListener(EventListener listener) {
+        listeners.add(listener);
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            executor.submit(() -> {
+                try {
+                    listener.onEvent(event);
+                } catch (Exception e) {
+                    throw new RuntimeException("Failed to handle an event" + event, e);
+                }
+            });
+        }
     }
 
     public void shutdown() {
-
+        executor.shutdown();
     }
```

Comment:
It's good that you're attempting to shut down the ExecutorService, but you should also consider calling 'awaitTermination' after 'shutdown' to ensure that all tasks have completed before the application continues. Additionally, you may want to handle the InterruptedException that 'awaitTermination' can throw.